tool/lib/commands/latest_flutter_candidate.dart

```diff
@@ -0,0 +1,277 @@
+import 'dart:convert';
+import 'dart:math' as math;
+
+import 'package:args/command_runner.dart';
+import 'package:http/http.dart' as http;
+
+/// Outputs the latest flutter candidate branch name.
+///
+/// The latest candidate branch will be the branch that matches the Flutter SDK
+/// inside g3.
+///
+/// Sample usage:
+/// ```shell
+///   $dart tool/bin/repo_tool.dart latest-flutter-candidate
+/// ```
+class LatestFlutterCandidateCommand extends Command {
+  // You can authorize your access if you run into a github rate limit.
+  // Don't check in your passwords or auth tokens.
+  static const auth = '';
+
+  @override
+  String get name => 'latest-flutter-candidate';
+
+  @override
+  String get description =>
+      'Outputs the most recent flutter release candidate banch';
+
+  @override
+  Future run() async {
+    SemanticVersion latest = SemanticVersion();
+    String? latestBranchName;
+
+    final allBranchNames = <String>[];
+
+    // TODO(kenz): consider traversing pages properly instead of using a while
+    // loop.
+    // See https://docs.github.com/en/rest/guides/traversing-with-pagination
+    const maxPerPage = 100;
+    var requestPage = 0;
+    bool lastPageReceived = false;
+    while (!lastPageReceived) {
+      final uri = Uri.https(
+        '${auth}api.github.com',
```

Comment:
i feel like this auth would need to be in a bearer token header

Reply:
you're right! this is what was causing the test flakes